.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,47 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
```

Comment:
you can keep linting to pull requests

Reply:
the --new-from-rev=origin/main option in our golangci-lint command, which inherently limits linting to only the new changes since the main branch. This approach makes the linting process efficient regardless of whether the workflow is triggered by a push or a pull_request.